### PR TITLE
Minor cleanup of editor and mapitems

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/layer_tiles.cpp [hunk 1]
```diff
@@ -177,6 +177,11 @@ void CLayerTiles::Clamp(RECTi *pRect)
 		pRect->w = 0;
 }
 
+bool CLayerTiles::IsGameLayer() const
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp:180]: While we're at it, I think the naming of this function is confusing. You could also expect it to match only against m_pGameLayer. I don't have a good alternative though.

Comment (Jupeyy) [reply src/game/editor/layer_tiles.cpp:180]: ingame we call it entities
IsEntitiesLayer

Or
IsGamePhysicsLayer 
xd

Comment (Robyt3) [reply src/game/editor/layer_tiles.cpp]: I renamed it to `IsEntitiesLayer`. We already had a `CEditor::IsSpecialLayer(CLayer *)` function, but I prefer the former name and also the more object oriented style of declaring the function on `CLayer`.

**Pull request merged**